src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveResourceService.cs

```diff
@@ -120,11 +124,12 @@ public async Task<RetrieveResourceResponse> GetInstanceResourceAsync(RetrieveRes
                     string.Format(CultureInfo.CurrentCulture, DicomCoreResource.RetrieveServiceMultiInstanceTranscodingNotSupported, requestedTransferSyntax));
             }
 
+            FileProperties fileProperties = await GetFileProperties(partition, instance, version, needsTranscoding, cancellationToken: cancellationToken);
```

Comment:
This is going to degrade in current retrieve performance if there is no transcoding. Are we okay with that?

Reply:
Same for frame resources too.

Reply:
If we don't do transcoding, we return the stream using DownloadStreamingAsync method, which may have Content.Length to get the whole file size. But may not work, could you check?